extract.py

```diff
@@ -42,27 +81,30 @@ def get_listing_data(page_num: int) -> list[dict]:
         date_tag = item.find("time", class_="judgment-listing__date")
 
         title = title_tag.get_text(strip=True) if title_tag else None
-        href = (
-            title_tag.find("a")["href"] if title_tag and title_tag.find("a") else None
-        )
+        href = title_tag.find("a")["href"] if title_tag and title_tag.find("a") else ""
         court = court_tag.get_text(strip=True) if court_tag else None
         citation = citation_tag.get_text(strip=True) if citation_tag else None
         date = date_tag.get("datetime") if date_tag else None
 
-        judgments.append(
-            {
+        if citation not in already_loaded:
+
+            html_data = {
                 "title": title,
-                "href": href,
+                "url": base_url + href,
                 "court": court,
                 "citation": citation,
                 "date": date,
-                "text_md": get_article_data(href)[0],
-                "text_raw": get_article_data(href)[1],
+                "text_raw": get_article_data(href),
             }
-        )
+            if validate_html_data(html_data):
+                judgments.append(html_data)
 
     return judgments
 
 
 if __name__ == "__main__":
```

Comment:
Potentially remember to remove this?

Reply:
I meant the things below this - the different urls tested

Reply:
this has been corrected, thanks